### PR TITLE
Make gravity game structure

### PR DESCRIPTION
**Pull request by sparshg**
Separated code into few files to make management easier.

https://user-images.githubusercontent.com/43041139/110253083-6d511700-7fae-11eb-8f4f-b204c5ddbfed.mp4

https://user-images.githubusercontent.com/43041139/110253086-6e824400-7fae-11eb-9ede-6706d18244cf.mp4

### CODE REVIEW

**Comment by MTM828** [comment]
Uum...
I've been working on it locally, this is the code:
```python
class Planet:
    def __init__(self, x, y, size):
        self.x = x
        self.y = y
        self.size = size


class Planets:
    def __init__(self):
        self.planets = [Planet(50, 150, 5), Planet(225, 300, 20), Planet(400, 75, 50)]

    def render(self):
        for i in range(len(self.planets)):
            pg.draw.circle(main.win, WHITE, (self.planets[i].x, self.planets[i].y), self.planets[i].size)


class Player:
    def __init__(self):
        self.x = 50
        self.y = 50
        self.xVel = 0
        self.yVel = 0

    def update(self):
        # Controls
        keyDown = pg.key.get_pressed()
        if keyDown[pg.K_RIGHT]:
            main.player.xVel += Physics.controlPower
        if keyDown[pg.K_LEFT]:
            main.player.xVel -= Physics.controlPower
        if keyDown[pg.K_UP]:
            main.player.yVel -= Physics.controlPower
        if keyDown[pg.K_DOWN]:
            main.player.yVel += Physics.controlPower

        # Gravity
        for i in range(len(main.planets.planets)):
            # Direction = atan2(destination y - y, destination x - x)
            # That finds the direction if you're trying to point somewhere (which is (in this case) "main.planets.planets[i]")
            dir = math.atan2(main.planets.planets[i].y - self.y, main.planets.planets[i].x - self.x)
            # Distance = hypot(x1 - x2, y1 - y2)
            # Finds the dist between 2 points (multiplied by Physics.gravity to adjust the vel)
            dist = math.hypot(self.x - main.planets.planets[i].x, self.y - main.planets.planets[i].y)

            self.xVel += math.cos(dir) * main.planets.planets[i].size / (dist * Physics.gravity)
            self.yVel += math.sin(dir) * main.planets.planets[i].size / (dist * Physics.gravity)

            # Bounce if touching planet
            if dist < main.planets.planets[i].size + 25:
                self.xVel *= -1
                self.yVel *= -1

            self.x += self.xVel * main.dt
            self.y += self.yVel * main.dt

        # Bounce if on edge
        if self.x - 25 < 0 or self.x + 25 > WIDTH:
            self.xVel *= -1
        if self.y - 25 < 0 or self.y + 25 > HEIGHT:
            self.yVel *= -1

    def render(self):
        pg.draw.circle(main.win, GREY, (self.x, self.y), 25)


class Physics:
    gravity = 0.5
    # How much planets attract the player (the bigger, the more they do)
    controlPower = 0.1
    # How powerful the controls of the player are


class Goal:
    def __init__(self, x, y, size):
        self.x = x
        self.y = y
        self.size = size
        self.particles = []

    def render(self):
        pg.draw.circle(main.win, YELLOW, (self.x, self.y), self.size)
        for i in range(len(self.particles)):
            self.particles[i].render()


    def confetti(self):
        for i in range(math.random() * 20):
            self.particles.append(Particle(main.goal.x, main.goal.y, random.randint(5, 10), math.random(0, 6.28318531)))


class Particle:
    def __init__(self, x, y, xVel, yVel, dir):
        self.x = x
        self.y = y
        self.xVel = xVel
        self.yVel = yVel
        self.dir = dir

    def update(self):
        pass

    def render(self):
        pg.draw.circle(main.win, YELLOW, (self.x, self.y), main.goal.size / 4)
```
There's more btw.
So... what am I supposed to do with this code? seperate some classes and create a bunch of files? (._. ')

**Comment by sparshg** [comment]
Just open a PR with your changes

**Review by MTM828 — APPROVED** [review]

**Review by MTM828 — APPROVED** [review]

File: src/gravity/universe.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        self.planets = [
+            Planet(10, 10, (WIDTH / 2 - 100, HEIGHT / 2), (0, 50)),
+            Planet(10, 10, (WIDTH / 2 + 100, HEIGHT / 2), (0, -50)),
+        ]
+
+    def update(self, dt):
+        for planet in self.planets:
+            planet.calcUpdate(self.planets)
+        for planet in self.planets:
+            planet.applyUpdate(dt)
```
Comment (sparshg) [inline src/gravity/universe.py:17]: notice I am using calcUpdate first to calculate the strength of gravitational field due to other planets. and then I applyUpdate to change positions of all planets to gether

Comment (MTM828) [reply src/gravity/universe.py:17]: Oh yeah, multiple for loops does sound like a better idea. 👍

**Pull request merged**